src/Linker.cs

```diff
@@ -0,0 +1,382 @@
+        /// </summary>
+        /// <param name="context">The store context that owns the instance.</param>
+        /// <param name="name">The name of the instance to define.</param>
+        /// <param name="instance">The instance to define.</param>
+        public void DefineInstance(StoreContext context, string name, Instance instance)
+        {
+            if (name is null)
+            {
+                throw new ArgumentNullException(nameof(name));
+            }
+
+            if (instance is null)
+            {
+                throw new ArgumentNullException(nameof(instance));
+            }
+
+            unsafe
+            {
+                var nameBytes = Encoding.UTF8.GetBytes(name);
+                fixed (byte* namePtr = nameBytes)
+                {
+                    var error = Native.wasmtime_linker_define_instance(handle, context.handle, namePtr, (UIntPtr)nameBytes.Length, instance.instance);
+                    if (error != IntPtr.Zero)
+                    {
+                        throw WasmtimeException.FromOwnedError(error);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Instantiates a module with imports from items defined in the linker.
+        /// </summary>
+        /// <param name="context">The store context to instantiate in.</param>
+        /// <param name="module">The module to instantiate.</param>
+        /// <returns>Returns the new instance.</returns>
+        public Instance Instantiate(StoreContext context, Module module)
+        {
+            if (module is null)
+            {
+                throw new ArgumentNullException(nameof(module));
+            }
+
+            var error = Native.wasmtime_linker_instantiate(handle, context.handle, module.NativeHandle, out var instance, out var trap);
+            if (error != IntPtr.Zero)
+            {
+                throw WasmtimeException.FromOwnedError(error);
+            }
+
+            if (trap != IntPtr.Zero)
+            {
+                throw TrapException.FromOwnedTrap(trap);
+            }
+
+            return new Instance(instance);
+        }
+
+        /// <summary>
+        /// Defines automatic instantiations of a module in this linker.
+        /// </summary>
+        /// <param name="context">The store context to instantiate in.</param>
+        /// <param name="module">The module to automatically instantiate.</param>
+        public void DefineModule(StoreContext context, Module module)
+        {
+            if (module is null)
+            {
+                throw new ArgumentNullException(nameof(module));
+            }
+
+            unsafe
+            {
+                var nameBytes = Encoding.UTF8.GetBytes(module.Name);
+                fixed (byte* namePtr = nameBytes)
+                {
+                    var error = Native.wasmtime_linker_module(handle, context.handle, namePtr, (UIntPtr)nameBytes.Length, module.NativeHandle);
```

Comment:
Technically I think this can do instantiation like above as well

Reply:
Opened issue #63 for this.